File: python/models/curso.py
from .programa_academico import ProgramaAcademico

class Curso(ProgramaAcademico):

    def __init__(self):
        super().__init__("Curso")

    def evaluar_aprobacion(self, notas):
        promedio = sum(notas) / len(notas)
        return promedio >= 10

File: java/Curso.java
import java.util.List;

public class Curso extends ProgramaAcademico {
    private final double notaMinima = 10.0;

    public Curso() {
        super("Curso");
    }

    @Override
    public boolean evaluarAprobacion(List<Double> notas) {
        return !notas.isEmpty() && notas.stream().mapToDouble(Double::doubleValue).average().orElse(0) >= notaMinima;
    }

    @Override
    public String obtenerCriterio() {
        return "promedio mayor o igual a 10/20";
    }
}
